plugin-docker/che-plugin-docker-client/src/main/java/org/eclipse/che/plugin/docker/client/DockerConnector.java

```diff
@@ -703,6 +715,77 @@ public ContainerProcesses top(String container, String... psArgs) throws IOExcep
     }
 
     /**
+     * Gets the files from the specified container.
+     *
+     * @param container
+     *         container id
+     * @param source
+     *         path to file or directory inside specified container
+     * @return stream of resources from the specified container filesystem, with retention connection
+     * @throws IOException
+     *         when problems occurs with docker api calls
+     * @apiNote this method implements 1.20 docker API and  requires docker not less than 1.8.* version
+     */
+    public InputStream getResourceFrom(String container, String source) throws IOException {
+        DockerConnection connection = openConnection(dockerDaemonUri);
+        final DockerResponse response = connection.method("GET")
+                                                  .path(String.format("/containers/%s/archive?path=%s", container, source))
+                                                  .request();
+        final int status = response.getStatus();
+        if (status != 200) {
+            final String msg = CharStreams.toString(new InputStreamReader(response.getInputStream()));
+            throw new DockerException(String.format("Error response from docker API, status: %d, message: %s", status, msg), status);
+        }
+
+        return new CloseConnectionInputStream(response.getInputStream(), connection);
+    }
+
+    /**
+     * Puts the files to the specified container.
+     *
+     * @param container
+     *         container id
+     * @param target
+     *         path to file or directory inside specified container
+     * @param is
+     *         stream of resources from filesystem
+     * @param noOverwrite
+     *         If "1", "true", or "True" then it will be an error
+     *         if unpacking the given content would cause an existing directory to be replaced
+     *         with a non-directory and vice versa.
+     * @throws IOException
+     *         when problems occurs with docker api calls, or during file system operations
+     * @apiNote this method implements 1.20 docker API and  requires docker not less than 1.8.* version
+     */
+    public void putResourceTo(String container, String target, InputStream is, boolean noOverwrite) throws IOException {
+        DockerConnection connection = openConnection(dockerDaemonUri);
+        File tarFile = Files.createTempFile("compressed-resources", ".tar").toFile();
+        OutputStream out = new BufferedOutputStream(new FileOutputStream(tarFile));
+        IOUtils.copy(is, out);
```

Comment:
Have you tried this native method instead of IOUtils:
http://docs.oracle.com/javase/7/docs/api/java/nio/file/Files.html#copy(java.io.InputStream,%20java.nio.file.Path,%20java.nio.file.CopyOption...) ?